extended-tracer/README.md

```diff
@@ -0,0 +1,149 @@
+# extended-tracer
+
+Utility methods to make it easier to use the OpenTelemetry tracer.
+
+## Usage Examples
+
+Here are some examples how the utility methods can help reduce boilerplate code.
+
+### Tracing a function
+
+Before:
+
+```java
+Span span = tracer.spanBuilder("reset_checkout").startSpan();
+String transactionId;
+try (Scope scope = span.makeCurrent()) {
+  transactionId = resetCheckout(cartId);
+} catch (Exception e) {
+  span.setStatus(StatusCode.ERROR);
+  span.recordException(e);
+  throw e;
+} finally {
+  span.end();
+}
+```
+
+After:
+
+```java
+Tracing tracing = new Tracing(openTelemetry, "service");
+String transactionId = tracing.call("reset_checkout", () -> resetCheckout(cartId));
```

Comment:
Firstly, thanks for putting together the examples @zeitlinger.

One question with this example, the checked exception is used to set an error on the span but the application code never sees it. Or are these usages targeted at instrumentation developers and not application developers?

Reply:
the exceptions should be re-thrown after being recorded, e.g. [here](https://github.com/zeitlinger/opentelemetry-java-contrib/blob/09c67e117fad6e44e48c10e9862e76473711a6b3/extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java#L90)

Reply:
Should we change `Exception` to `Throwable` in the example (and in the `Tracing` class)?

Reply:
Could the example be updated to show application code needing to change from catching `Exception` to `Throwable`? It's an important point to note as developers will need to be aware the checked exception their code would normally throw is being hidden in a Throwable.

Reply:
good points - both of them :smile: 

I changed to use `Throwable` consistently & added a separate section to explain error handling